store/search/ts.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/typesense/typesense-go/typesense"
+	tsApi "github.com/typesense/typesense-go/typesense/api"
+)
+
+type storeImpl struct {
+	client *typesense.Client
+}
+
+type IndexDocumentsOptions struct {
+	Action    string
+	BatchSize int
+}
+
+func (s *storeImpl) convertToInternalError(err error) error {
+	if e, ok := err.(*typesense.HTTPError); ok {
```

Comment:
```
var ep typesense.HTTPError
if errors.As(err, &ep) {
....
}
```

Reply:
`As` is panicking, reverting it back to check explicitly for type.

Reply:
Interesting, ok.